app/src/main/res/layout/fragment_login.xml

```diff
@@ -0,0 +1,101 @@
+<?xml version="1.0" encoding="utf-8"?>
+<androidx.constraintlayout.widget.ConstraintLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:app="http://schemas.android.com/apk/res-auto"
+    xmlns:tools="http://schemas.android.com/tools"
+    android:layout_width="match_parent"
+    android:layout_height="match_parent"
+    android:gravity="center"
+    android:orientation="vertical">
+
+    <ImageView
+        android:id="@+id/vHeaderImage"
```

Comment:
Maybe for all of these Ids you may want to rename them to add the Login word in their name. If you have a name that is something like "vMainButton" (as an example), if you have a main button in a lot of views, you will not be able to identify that main button. The best way is call this this like this: vLoginHeaderImage, vLoginMainButton, vLoginEmail, vLoginPassword. Another thing is that it is not always useful to have the type of view in the name. For example, vLoginEmailInput. You already know it is an input, an it is the only reference to a email.